### ISSUE TITLE
Last } is not on its own line for JSX

### ISSUE THREAD

**Issue by chrisdrackett** [prettier/prettier#765]
I sometimes use the following format in JSX:

```jsx
{booleanValue &&
    <div>only show this when booleanValue is true</div>
}
```

currently prettier outputs this in the following way:

```jsx
{booleanValue &&
    <div>only show this when booleanValue is true</div>}
```

IMO I think the output should stay in the initial format.

**Comment by aaronjensen** [comment]
for what it's worth, I've seen the latter way more often in examples/actual code.

I wouldn't consider the former an option just because it sort of convolutes two bracket styles. I'd say it's either your latter or:

```jsx
{
  booleanValue &&
    <div>...</div>
}
```

but that's unnecessarily verbose. In short, I'd say it's right as-is.

**Comment by chrisdrackett** [comment]
to me this feels very similar to the `jsxBracketSameLine` option. I'm not against the above, but I personally prefer my initial format as its both somewhat compact but more readable than the current output.

**Comment by aaronjensen** [comment]
It may feel similar, but it's quite different. w/ `jsxBracketSameLine` it's still working w/ opening and closing markers. `<div` is the opening marker and `>` is the closing marker, so the question is do you put the closing marker on its own line or the end of the last line. In both cases, you put the opening marker on its own line.

You are proposing to put the opening marker `{` on a shared line and the closing marker `}` on its own line.

This is tantamount to these:

```js
const foo = () => {const x = 3;
  return x;
}

render() {return (<div>
      hello!
    </div>
  );
}

**Comment by vjeux** [comment]
We've had this discussion in a pull request and people slightly preferred your suggestion but given the way prettier is currently implemented, it is not trivial to write it correctly so we settled on the second version.

If someone is motivated, that could be a valuable addition.

**Comment by chrishoage** [comment]
For what it's worth, I find the current formatting very difficult to read

```
{booleanValue &&
    <div>only show this when booleanValue is true</div>}
```
vs

```
{
  booleanValue &&
    <div>only show this when booleanValue is true</div>
}
```

The latter is much much easier to read, the brackets don't get lost. This is particularly pronounced when a map is conditionally rendered. 

```
      {things.map(
        thing =>
          reallyLongPleaseDontPutOnOneLine(thing)
            ? <p>
                {ok}
              </p>
            : <Quax bar={bar} baz={baz} {...thing} />
      )}
```

This is more difficult to read than 

```
      {
        things.map(
          thing =>
            reallyLongPleaseDontPutOnOneLine(thing)
              ? <p>
                  {ok}
                </p>
              : <Quax bar={bar} baz={baz} {...thing} />
        )
      }
```

Please consider changing this for Prettier 2.0, it is the only thing keeping me from using it for JSX projects.

**Comment by rattrayalex** [comment]
Similar behavior has been merged in #2398 , closing
